### ISSUE TITLE
Return code 101 

### ISSUE THREAD

**Issue by ccwanggl** [neocmakelsp/neocmakelsp#119]
The latest version 0.8.10 with error as below

[ERROR][2024-10-11 07:20:30] .../vim/lsp/rpc.lua:770	"rpc"	"C:\\Users\\cnm2020\\AppData\\Local\\nvim-data\\mason\\bin\\neocmakelsp.CMD"	"stderr"	"2024-10-10T23:20:30.077130Z  INFO neocmakelsp::languageserver: find cache-v2 json, start reading the data\n2024-10-10T23:20:30.077219Z  INFO neocmakelsp::languageserver: Finish getting the data in cache-v2 json\n"
[ERROR][2024-10-11 07:20:37] .../vim/lsp/rpc.lua:770	"rpc"	"C:\\Users\\cnm2020\\AppData\\Local\\nvim-data\\mason\\bin\\neocmakelsp.CMD"	"stderr"	"thread 'main' panicked at C:\\Users\\cnm2020\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\neocmakelsp-0.8.10\\src\\jump.rs:518:66:\ncalled `Result::unwrap()` on an `Err` value: ()\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"

**Comment by Decodetalkers** [comment]
can you print the local_path in 518? on your machine? I do not have windows machine. I think there maybe some problem with the path

**Comment by Decodetalkers** [comment]
https://github.com/neocmakelsp/neocmakelsp/commit/dde819917cb33759ef4246531aa777fe676f7c6a can you help check if this commit fix your problem?

**Comment by ccwanggl** [comment]
sorry, i am not familer with rust, dose tcp mode get detail info?

**Comment by Decodetalkers** [comment]
> sorry, i am not familer with rust, dose tcp mode get detail info?

Use tcp mode to start neocmakelsp, and see if it works now. Thanks for your help

**Comment by ccwanggl** [comment]
![image](https://github.com/user-attachments/assets/19ceb12f-3051-405e-80d3-e112d692c6f7)

![image](https://github.com/user-attachments/assets/53981e93-e8c1-47e0-b952-0611f1bbb956)


Seems work, right?

**Comment by Decodetalkers** [comment]
> ![image](https://github.com/user-attachments/assets/19ceb12f-3051-405e-80d3-e112d692c6f7)
> 
> ![image](https://github.com/user-attachments/assets/53981e93-e8c1-47e0-b952-0611f1bbb956)
> 
> 
> Seems work, right?

Does package completing ,jump work well?

**Comment by ccwanggl** [comment]
> > ![image](https://private-user-images.githubusercontent.com/8803306/375678930-19ceb12f-3051-405e-80d3-e112d692c6f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2MzU3NDcsIm5iZiI6MTcyODYzNTQ0NywicGF0aCI6Ii84ODAzMzA2LzM3NTY3ODkzMC0xOWNlYjEyZi0zMDUxLTQwNWUtODBkMy1lMTEyZDY5MmM2ZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTFUMDgzMDQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWUwY2U0NjhhOWVhNmM5OTcyNGRiNDdhZWJhNGJhNGM1OWQyNDNlZjljM2JkNzY4YTBmYjY1NDlkYjE1Yjc4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.j2qC4qcdQPVOV3BzI4ft4O9MvDGhzorx4M7cGlpU1Io)
> > ![image](https://private-user-images.githubusercontent.com/8803306/375679029-53981e93-e8c1-47e0-b952-0611f1bbb956.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2MzU3NDcsIm5iZiI6MTcyODYzNTQ0NywicGF0aCI6Ii84ODAzMzA2LzM3NTY3OTAyOS01Mzk4MWU5My1lOGMxLTQ3ZTAtYjk1Mi0wNjExZjFiYmI5NTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTFUMDgzMDQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTliYTJiOTJlYTA2NGU1Y2YzZTVhYjcyZjRhMmQ3YTg2NTIxY2JmMTU3MWQyNDgyOTg2ZmNiODc3Yjc1ODkzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SjIGTsbujAz0dWMKIkXxXfoGS8IgrmUl4I1xfIeZojM)
> > Seems work, right?
> 
> Does package completing ,jump work well? Also try using the add_subdirectory function, and to see if the editor can jump to the subdirectory, and see if the manually created function can be completed successfully

Yes, I think every thing work fine.

**Comment by ccwanggl** [comment]
But stdio mode not work? you can tell me how to get more info in stdio mode if possible, i will happy to test

**Comment by Decodetalkers** [comment]
> But stdio mode not work? you can tell me how to get more info in stdio mode if possible, i will happy to test

Tcp mode works the same with stdio mode. If you want to use stdio mode, run cargo build --release, and copy the target to the place where mason run it

**Comment by ccwanggl** [comment]
> > But stdio mode not work? you can tell me how to get more info in stdio mode if possible, i will happy to test
> 
> Tcp mode works the same as stdio mode. If you want to use stdio mode, run cargo build --release, and copy the target to the place where mason run it

Yes, i have done those steps, and work fine, I think you can release current codebase, Thanks!

**Comment by Decodetalkers** [comment]
Thanks for your efforts, I will close this issue now
